diff --git a/mt_wifi/embedded/common/cmm_cfg.c b/mt_wifi/embedded/common/cmm_cfg.c
index a3482fa..24367ee 100644
--- a/mt_wifi/embedded/common/cmm_cfg.c
+++ b/mt_wifi/embedded/common/cmm_cfg.c
@@ -907,7 +907,7 @@ INT RtmpIoctl_rt_ioctl_giwname(
 					return NDIS_STATUS_FAILURE;
 				}
 			} else {
-				strlcpy(pData, "RTWIFI SoftAP", strlen("RTWIFI SoftAP") + 1);
+				strscpy(pData, "RTWIFI SoftAP", strlen("RTWIFI SoftAP") + 1);
 			}
 		}
 	}
diff --git a/mt_wifi/embedded/common/cmm_info.c b/mt_wifi/embedded/common/cmm_info.c
index 2b192e9..219e33d 100644
--- a/mt_wifi/embedded/common/cmm_info.c
+++ b/mt_wifi/embedded/common/cmm_info.c
@@ -16269,7 +16269,7 @@ INT Set_TxBfProfileDataWrite(RTMP_ADAPTER *pAd, RTMP_STRING *arg)
 		strLen = strlen(value);
 
 		if (strLen & 1) {
-			strlcpy(value_T, "0", sizeof(value_T));
+			strscpy(value_T, "0", sizeof(value_T));
 			strncat(value_T, value, strLen);
 			AtoH(value_T, (PCHAR)(&Input[i]), 2);
 			Input[i] = be2cpu16(Input[i]);
@@ -16431,7 +16431,7 @@ between : */
 		str_len = strlen(value);
 
 		if (str_len & 1) {
-			strlcpy(value_t, "0", sizeof(value_t));
+			strscpy(value_t, "0", sizeof(value_t));
 			strncat(value_t, value, str_len);
 			AtoH(value_t, (PCHAR)(&input[i]), 2);
 			input[i] = be2cpu16(input[i]);
@@ -16601,7 +16601,7 @@ INT Set_TxBfProfilePnWrite(RTMP_ADAPTER *pAd, RTMP_STRING *arg)
 		strLen = strlen(value);
 
 		if (strLen & 1) {
-			strlcpy(value_T, "0", sizeof(value_T));
+			strscpy(value_T, "0", sizeof(value_T));
 			strncat(value_T, value, strLen);
 			AtoH(value_T, (PCHAR)(&Input[i]), 2);
 			Input[i] = be2cpu16(Input[i]);
diff --git a/mt_wifi/embedded/include/meminfo_list.h b/mt_wifi/embedded/include/meminfo_list.h
index 4b5eaf5..b68995b 100644
--- a/mt_wifi/embedded/include/meminfo_list.h
+++ b/mt_wifi/embedded/include/meminfo_list.h
@@ -344,7 +344,7 @@ static inline VOID MIListAddHead(MEM_INFO_LIST *MIList, UINT32 Size, VOID *pMemA
 	pEntry->Line = Line;
 	pEntry->pLastPos = 0;
 	pEntry->uPosIdx = 0;
-	strlcpy(pEntry->Func, Func, sizeof(pEntry->Func));
+	strscpy(pEntry->Func, Func, sizeof(pEntry->Func));
 	pheadMemEntry = &MIList->pMemAddr[MemIndex];
 	pheadCalEntry = &MIList->pCaller[CallerIdx];
 	
